carla/recourse_methods/catalog/dice/model.py

```diff
@@ -0,0 +1,83 @@
+import dice_ml
+import pandas as pd
+
+from ...api import Recourse_Method
+
+
+class Dice(Recourse_Method):
+    def __init__(self, mlmodel, data):
+        """
+        Constructor for Dice model
+        Implementation can be seen at https://github.com/interpretml/DiCE
+
+        Restrictions:
+        ------------
+        -   Only the model agnostic approach (backend: sklearn) is used in our implementation.
+        -   ML model needs to have a transformation pipeline for normalization, encoding and feature order.
+            See pipelining at carla/models/catalog/catalog.py for an example ML model class implementation
+
+        Parameters
+        ----------
+        mlmodel : models.api.MLModel
+            ML model to build counterfactuals for.
+        data : data.api.Data
+            Underlying dataset we want to build counterfactuals for.
+        """
+        # Prepare data for dice data structure
+        self._dice_data = dice_ml.Data(
+            dataframe=data.raw,
+            continuous_features=data.continous,
+            outcome_name=data.target,
+        )
+
+        # Build dice model structure
+        # Since our own model class MLModel resembles the sklearn structure it may suffice to use the sklearn
+        # backend implementation made by dice
+        self._backend = "sklearn"
+
+        self._dice_model = dice_ml.Model(model=mlmodel, backend=self._backend)
+
+        self._dice = dice_ml.Dice(self._dice_data, self._dice_model, method="random")
+
+    @property
+    def dice_model(self):
+        return self._dice
+
+    def get_counterfactuals(self, factuals, num_of_cf, desired_class):
```

Comment:
I think `num` would suffice since we also have the docstring to explain further

Reply:
Will be changed :+1: